crates/rpc/src/methods/send_transaction.rs

```diff
@@ -189,12 +165,26 @@ impl SendTransaction {
             .parse()
             .map_err(|_| Error::CannotParseUrl(self.network.http_url.clone()))?;
 
-        let provider = ProviderBuilder::new()
-            .with_recommended_fillers()
-            .wallet(signer.to_wallet())
-            .on_http(url);
+        if self.network.is_dev() {
+            if let Wallet::Impersonator(ref wallet) = wallet {
+                let account = wallet.get_address(&self.wallet_path).await?;
+
+                let provider = ProviderBuilder::new()
+                    .with_recommended_fillers()
+                    .on_http(url);
+
+                provider.anvil_impersonate_account(account).await?;
+
+                self.provider = Some(Box::new(provider));
+            }
+        } else {
+            let provider = ProviderBuilder::new()
+                .with_recommended_fillers()
+                .wallet(signer.to_wallet())
+                .on_http(url);
+            self.provider = Some(Box::new(provider));
+        }
```

Comment:
this whole block seems like it could be simplified

```rust
let builder = ProviderBuilder::new()
  .with_recommended_fillers();

let builder = if ... {
  builder.anvil_impersonate_account(account);
  builder
} else {
  builder.wallet(signer.to_wallet())
}

self.provider = Some(Box::new(builder.on_http(url)));
```

Reply:
Looks like it but we run into the same issues as before we switched to a `Box<dyn Provider<...>>`. I can only call `anvil_impersonate_account` on a `ProviderBuilder` that starts with an `Identity` type. I could move the `on_http` call to immediately after `with_recommended_fillers` to fix that but then the `builder.wallet` breaks as it needs a `FillProvider`